### PR TITLE
[297] Improve user index

### LINKED ISSUE #297: Improve User Index
- [x] The lists should be tables
- [x] We should show netIDs

### PR DESCRIPTION
**Pull request by orenyk**
Resolves #297
- Add users/table and users/actions partials
- Add UsersController#destroy with feature specs

Blocked on #397 (for user deletion spec)

### CODE REVIEW

**Comment by esoterik** [comment]
@orenyk can you rebase?

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,41 +1,24 @@
 <h1>Users</h1>
-<h3>Admins</h3>
+<h3 id="admins">Admins</h3>
+<p><a href="#reps">Housing Committee</a> / <a href="#students">Students</a></p>
```
Comment (esoterik) [inline app/views/users/index.html.erb]: can we add the jumping nav to the top of the page and have a "Back to top" link under each header instead?

Comment (orenyk) [inline app/views/users/index.html.erb]: Sure that's fine. You may not like my fix to #306 in #385 though 😕

**Comment by orenyk** [comment]
@esoterik back to you!

**Comment by orenyk** [comment]
@esoterik fixed and squashed. Good to merge?

**Pull request merged**